pandas/io/sas/sas7bdat.py

```diff
@@ -703,15 +705,38 @@ def _chunk_to_dataframe(self):
                 rslt[name] = self._byte_chunk[jb, :].view(dtype=self.byte_order + "d")
                 rslt[name] = np.asarray(rslt[name], dtype=np.float64)
                 if self.convert_dates:
-                    unit = None
+                    warn_msg = (
+                        "date > pandas.Timestamp.max, returning "
+                        "datetime.datetime objects instead"
+                    )
                     if self.column_formats[j] in const.sas_date_formats:
-                        unit = "d"
+                        try:
```

Comment:
These try / excepts are likely complicated enough to warrant a refactor into a standalone function.